PerfTools/AllocMonitorPreload/src/memory_proxies.cc

```diff
@@ -0,0 +1,419 @@
+    return original(nitems, item_size);
+  }
+  auto& reg = AllocMonitorRegistry::instance();
+  return reg.allocCalled(
+      nitems * item_size,
+      [nitems, item_size, original]() { return original(nitems, item_size); },
+      [](auto ret) { return malloc_usable_size(ret); });
+}
+#else
+void* malloc(size_t size) noexcept {
+  static const auto original = get<decltype(&::malloc)>("malloc");
+  if (not alloc_monitor_running_state()) {
+    return original(size);
+  }
+  auto& reg = AllocMonitorRegistry::instance();
+  return reg.allocCalled(
+      size, [size]() { return original(size); }, [](auto ret) { return malloc_usable_size(ret); });
+}
+
+void* calloc(size_t nitems, size_t item_size) noexcept {
+  static const auto original = get<decltype(&::calloc)>("calloc");
+  if (not alloc_monitor_running_state()) {
+    return original(nitems, item_size);
+  }
+  auto& reg = AllocMonitorRegistry::instance();
+  return reg.allocCalled(
+      nitems * item_size,
+      [nitems, item_size]() { return original(nitems, item_size); },
+      [](auto ret) { return malloc_usable_size(ret); });
+}
+#endif
+
+void* realloc(void* ptr, size_t size) noexcept {
+  static auto original = get<decltype(&::realloc)>("realloc");
+  if (not alloc_monitor_running_state()) {
+    return original(ptr, size);
+  }
+  size_t oldsize = malloc_usable_size(ptr);
+  void* ret;
+  auto& reg = AllocMonitorRegistry::instance();
+  {
+    //incase this calls malloc/free
+    [[maybe_unused]] auto g = reg.makeGuard();
+    ret = original(ptr, size);
+  }
+  size_t used = malloc_usable_size(ret);
+  if (used != oldsize) {
+    reg.deallocCalled([]() {}, [oldsize]() { return oldsize; });
+    reg.allocCalled(
+        size, []() { return nullptr; }, [used](auto) { return used; });
+  }
+  return ret;
+}
+
+void* aligned_alloc(size_t alignment, size_t size) noexcept {
+  static auto original = get<decltype(&::aligned_alloc)>("aligned_alloc");
+  if (not alloc_monitor_running_state()) {
+    return original(alignment, size);
+  }
+
+  auto& reg = AllocMonitorRegistry::instance();
+  return reg.allocCalled(
+      size,
+      [alignment, size]() { return original(alignment, size); },
+      [](auto ret) { return malloc_usable_size(ret); });
+}
+
+void free(void* ptr) noexcept {
+  static auto original = get<decltype(&::free)>("free");
+  // ignore memory allocated from our static array at startup
+  if (not is_local_alloc(ptr)) {
+    if (not alloc_monitor_running_state()) {
+      original(ptr);
+      return;
+    }
+
+    auto& reg = AllocMonitorRegistry::instance();
+    reg.deallocCalled([ptr]() { original(ptr); }, [ptr]() { return malloc_usable_size(ptr); });
+  }
+}
+}  // extern "C"
+
+//----------------------------------------------------------------
+//C++ memory functions
+
+#define CPP_MEM_OVERRIDE
+
+#if defined(CPP_MEM_OVERRIDE)
+#include <new>
+
+void* operator new(std::size_t size) {
+  static auto original = get<void* (*)(std::size_t)>("_Znwm");
```

Comment:
```suggestion
  static const auto original = get<void* (*)(std::size_t)>("_Znwm");
```
and hereafter (I won't repeat the comment more)?